res/controllers/Traktor-Kontrol-S4-MK3-hid-scripts.js

```diff
@@ -0,0 +1,884 @@
+    this.addButton(m1, "left", "fx2_on", 0x02, 0x80);
+    this.addButton(m1, "left", "fx3_on", 0x02, 0x08);
+    this.addButton(m1, "left", "fx4_on", 0x02, 0x04);
+
+    this.addButton(m1, "deckC", "ext", 0x03, 0x02);
+    this.addButton(m1, "deckA", "fx_assign_prev", 0x03, 0x08);
+    this.addButton(m1, "deckA", "fx_assign_next", 0x03, 0x10);
+    this.addButton(m1, "deckB", "fx_assign_prev", 0x03, 0x20);
+    this.addButton(m1, "deckB", "fx_assign_next", 0x03, 0x40);
+    this.addButton(m1, "deckC", "fx_assign_prev", 0x03, 0x01);
+    this.addButton(m1, "deckC", "fx_assign_next", 0x03, 0x04);
+    this.addButton(m1, "deckD", "fx_assign_prev", 0x03, 0x80);
+    
+    this.addButton(m1, "left", "pad1", 0x04, 0x20);
+    this.addButton(m1, "left", "pad2", 0x04, 0x10);
+    this.addButton(m1, "left", "pad3", 0x04, 0x80);
+    this.addButton(m1, "left", "pad4", 0x04, 0x40);
+    this.addButton(m1, "left", "pad5", 0x04, 0x08);
+    this.addButton(m1, "left", "pad6", 0x04, 0x04);
+    this.addButton(m1, "left", "pad7", 0x04, 0x02);
+    this.addButton(m1, "left", "pad8", 0x04, 0x01);
+    
+    this.addButton(m1, "left", "play_pause", 0x05, 0x01);
+    this.addButton(m1, "left", "cue", 0x05, 0x02);
+    this.addButton(m1, "left", "pad_hotcues", 0x05, 0x04);
+    this.addButton(m1, "left", "pad_record", 0x05, 0x08);
+    this.addButton(m1, "left", "pad_samples", 0x05, 0x10);
+    this.addButton(m1, "left", "pad_mute", 0x05, 0x20);
+    
+    this.addButton(m1, "left", "pad_stems", 0x06, 0x01);
+    this.addButton(m1, "left", "shift", 0x06, 0x02);
+    this.addButton(m1, "left", "select_deck_A", 0x06, 0x04);
+    this.addButton(m1, "left", "select_deck_C", 0x06, 0x08);
+    this.addButton(m1, "left", "jog_mode", 0x06, 0x10);
+    this.addButton(m1, "left", "tt_mode", 0x06, 0x20);
+    this.addButton(m1, "left", "grid_mode", 0x06, 0x40);
+    this.addButton(m1, "left", "sync_lock", 0x06, 0x80);
+
+    this.addButton(m1, "left", "move_encoder_press", 0x07, 0x04);
+    this.addButton(m1, "left", "move_encoder_touch", 0x07, 0x08);
+    this.addButton(m1, "left", "loop_encoder_touch", 0x07, 0x10);
+    this.addButton(m1, "left", "loop_encoder_press", 0x07, 0x20);
+
+    this.addButton(m1, "deckA", "mixerfx_on", 0x08, 0x01);
+    this.addButton(m1, "deckB", "mixerfx_on", 0x08, 0x20);
+    this.addButton(m1, "deckC", "mixerfx_on", 0x08, 0x02);
+    this.addButton(m1, "deckD", "mixerfx_on", 0x08, 0x10);
+    this.addButton(m1, "deckA", "headphone_cue", 0x08, 0x08);
+    this.addButton(m1, "deckB", "headphone_cue", 0x08, 0x40);
+    this.addButton(m1, "deckC", "headphone_cue", 0x08, 0x04);
+    this.addButton(m1, "deckD", "headphone_cue", 0x08, 0x80);
+
+    this.addButton(m1, "mixer", "select_fx_1", 0x09, 0x20);
+    this.addButton(m1, "mixer", "select_fx_2", 0x09, 0x02);
+    this.addButton(m1, "mixer", "select_fx_3", 0x09, 0x40);
+    this.addButton(m1, "mixer", "select_fx_4", 0x09, 0x01);
+    this.addButton(m1, "mixer", "select_fx_filter", 0x09, 0x80);
```

Comment:
We could target this PR to the main branch for 2.4 so it can use #2618. Alternatively we could leave these unmapped for a 2.3 mapping and add those features in the main branch.